e2e/cypress/integration/enterprise/system_console/download_compliance_export_file_spec.js

```diff
@@ -0,0 +1,213 @@
+        cy.get('input[data-testid="FileSettings.AmazonS3AccessKeyIdinput"]').type(AWS_ACCESS_KEY_ID);
+        cy.get('input[data-testid="FileSettings.AmazonS3SecretAccessKeyinput"]').type(AWS_SECRET_ACCESS_KEY);
+
+        // # Save file storage settings
+        cy.get('button[data-testid="saveSetting"]').click();
+
+        waitUntilConfigSave();
+
+        // # Goto compliance page and enable export
+        gotoCompliancePage();
+        enableComplianceExport();
+
+        // # Navigate to a team and post an attachment
+        gotoTeamAndPostImage();
+
+        // # Goto compliance page and start export
+        gotoCompliancePage();
+        exportCompliance();
+
+        // # Get the download link
+        cy.get('@firstRow').findByText('Download').parents('a').should('exist').then((fileAttachment) => {
+            const fileURL = fileAttachment.attr('href');
+
+            // # Download the file
+            downloadAttachmentAndVerifyItsProperties(fileURL);
+        });
+    });
+});
+
+function enableComplianceExport() {
+    // # Enable compliance export
+    cy.get('input[data-testid="enableComplianceExporttrue"]').click();
+
+    // # Change export format to CSV
+    cy.get('select[data-testid="exportFormatdropdown"]').select('csv');
+
+    // # Save settings
+    cy.get('button[data-testid="saveSetting"]').click();
+
+    waitUntilConfigSave();
+}
+
+function gotoCompliancePage() {
+    cy.visit('/admin_console/compliance/export');
+    cy.get('.admin-console__header', {timeout: TIMEOUTS.ONE_MIN}).should('be.visible').invoke('text').should('include', 'Compliance Export');
+}
+
+function gotoTeamAndPostImage() {
+    // # Get user teams
+    cy.apiGetTeamsForUser().then(({teams}) => {
+        const team = teams[0];
+        cy.visit(`/${team.name}/channels/town-square`);
+        cy.get('#post_textbox', {timeout: TIMEOUTS.ONE_MIN}).should('be.visible');
+    });
+
+    // # Remove images from post message footer if exist
+    cy.waitUntil(() => cy.get('#postCreateFooter').then((el) => {
+        if (el.find('.post-image.normal').length > 0) {
+            cy.get('.file-preview__remove > .icon').click();
+        }
+        return el.find('.post-image.normal').length === 0;
+    }));
+    const file = {
+        filename: 'image-400x400.jpg',
+        originalSize: {width: 400, height: 400},
+        thumbnailSize: {width: 400, height: 400},
+    };
+    cy.get('#fileUploadInput').attachFile(file.filename);
+
+    // # Wait until the image upload process start
+    cy.waitUntil(() => {
+        return Cypress.$('.post-image__uploadingTxt').length;
+    });
+
+    // # Wait until the image is uploaded
+    cy.waitUntil(() => {
+        return !Cypress.$('.post-image__uploadingTxt').length;
+    }, {
+        timeout: TIMEOUTS.FIVE_MIN,
+        interval: TIMEOUTS.ONE_SEC,
+        errorMsg: 'Unable to upload attachment in time',
+    });
+
```

Comment:
@josephbaylon, Initially I have written a code similar to this and it was giving "Error: Timed out retrying". That's why I have written 2 checks instead of one.

First one to check for "upload start"
Second one to check for "upload completed"

I thought your change might fix it but I'm still getting the same error.

Reply:
Also this is just failing with S3. That's why I have put this check as it was taking more time for me while uploading to S3.